tutorials/flume-capture-events.md

```diff
@@ -0,0 +1,179 @@
+---
+layout: page
+title: Capturing Events with Flume
+---
+
+Once you have an [Events dataset][events], you can create a web application that captures session events.
+
+This example shows how you can send log information via Flume to your Hadoop database using a JSP and custom servlet running on Tomcat.
+
+[events]:{{site.baseurl}}/tutorials/create-events-dataset.html
+
+## Configuring Flume
+
+These are the steps to configure Flume to channel log information directly to the `events` dataset. You first generate the configuration information using the Kite command-line interface, copy the results, paste them in the Flume configuration file, and then restart Flume.
+
+1. In a terminal window, type `kite-dataset flume-config --channel-type memory events`.
+1. Copy the output from the terminal window.
+1. Open Cloudera Manager.
+1. Under __Status__, click the link to __Flume__.
+1. Choose the __Configuration__ tab.
+1. Click __Agent Base Group__.
+1. Right-click the Configuration File text area and choose __Select All__.
+1. Right-click the Configuration File text area and choose __Paste__.
+1. Click __Save Changes__.
+1. From the __Actions__ menu, choose __Restart__, and confirm the action.
+
+Flume is now configured to receive logging events and record them in the `events` dataset.
+
+## Creating Web Application Pages
```

Comment:
This has a lot of discussion on the servlet and JSP, but the focus should be on the architecture: The servlet (or any application) uses Log4j to "log" events, which are Avro records. Log4j is configured to send those events to Flume. Flume accumulates events and writes them to the Dataset.

Most people want to know  _why_ we are doing it this way:

1. Why Flume? Because it is reliable, widely used for this purpose, and has good integration.
2. Why not write directly to the dataset in Hadoop? Because we want to accumulate events to write into the same file without disrupting the web application. Flume will take responsibility for individual events and see that they are written in groups to the dataset. The application wants to move on after a single event.